spec/kracken/authenticator_spec.rb

```diff
@@ -0,0 +1,62 @@
+require 'spec_helper'
+
+module Kracken
+  describe Authenticator do
+
+
+    describe ".with_credentials" do
+      it "returns nil when nothing is found" do
+        expect_any_instance_of(CredentialAuthenticator)
```

Comment:
I know, I am open to suggestions. Torn between injection and configuration vs. complexity.